executor/builder.go

```diff
@@ -1062,20 +1062,34 @@ func (b *executorBuilder) buildAnalyze(v *plan.Analyze) Executor {
 	return e
 }
 
-func (b *executorBuilder) constructDAGReq(plans []plan.PhysicalPlan) (*tipb.DAGRequest, error) {
+func (b *executorBuilder) constructDAGReq(plans []plan.PhysicalPlan) (*tipb.DAGRequest, bool, error) {
 	dagReq := &tipb.DAGRequest{}
 	dagReq.StartTs = b.getStartTS()
 	dagReq.TimeZoneOffset = timeZoneOffset(b.ctx)
 	sc := b.ctx.GetSessionVars().StmtCtx
 	dagReq.Flags = statementContextToFlags(sc)
+	streaming := true
 	for _, p := range plans {
 		execPB, err := p.ToPB(b.ctx)
 		if err != nil {
-			return nil, errors.Trace(err)
+			return nil, false, errors.Trace(err)
+		}
+		if !supportStreaming(p) {
+			streaming = false
 		}
 		dagReq.Executors = append(dagReq.Executors, execPB)
 	}
-	return dagReq, nil
+	return dagReq, streaming, nil
+}
+
+func supportStreaming(p plan.PhysicalPlan) bool {
+	switch p.(type) {
+	case *plan.PhysicalTableScan, *plan.PhysicalIndexScan:
+		return true
+	case *plan.PhysicalSelection, *plan.PhysicalProjection, *plan.PhysicalLimit:
```

Comment:
What about stream aggregation?

Reply:
Don't need it to support it, the result will not be too large.

Reply:
I think hash and merge aggregation can also be streamed.

Reply:
@lamxTyler Limit can use streaming, because unlike TopN, the chunk and range mapping can be retained.
@coocood Aggregation result could be large, if the column have too many distinguished value.
@zz-jason We may support coprocessor streaming aggregation later, after a thoughtful consideration about it, but just leave it disabled now.

Reply:
@lamxTyler But it seems useless... I'll disable it.